mmtf-codec/src/test/java/org/rcsb/mmtf/serialization/mp/MessagePackTest.java

```diff
@@ -0,0 +1,123 @@
+package org.rcsb.mmtf.serialization.mp;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.StringTokenizer;
+
+import junit.framework.TestCase;
+import org.rcsb.mmtf.api.StructureDataInterface;
+import org.rcsb.mmtf.dataholders.MmtfStructure;
+import org.rcsb.mmtf.decoder.GenericDecoder;
+import org.rcsb.mmtf.decoder.ReaderUtils;
+import org.rcsb.mmtf.serialization.MessagePackSerialization;
+
+import org.unitils.reflectionassert.ReflectionAssert;
+
+/**
+ * Tests if both MessagePack decoding methods, Jackson and the manual object
+ * creation gives identical data.
+ *
+ */
+public class MessagePackTest extends TestCase {

```

Comment:
Better use junit4. It doesn't require extending TestCase, just adding the \@Test annotation to the method you want to make into a test